packages/grafana-ui/src/utils/processTimeSeries.ts

```diff
@@ -173,3 +190,38 @@ export function processTimeSeries({ timeSeries, nullValueMode }: Options): TimeS
 
   return vmSeries;
 }
+
+export const toTableData = (results: any[]): TableData[] => {
+  const tables: TableData[] = [];
+  if (results) {
+    for (let i = 0; i < results.length; i++) {
+      const data = results[i];
+      if (data) {
+        if (data.hasOwnProperty('columns')) {
+          tables.push(data as TableData);
+        } else if (data.hasOwnProperty('datapoints')) {
+          const ts = data as TimeSeries;
+          tables.push({
+            type: 'timeseries',
```

Comment:
I'd like to see this property type guarded slightly better, as it's a `string` now. What are other types that can land in this property?

Reply:
Even better... we can just drop the type field.  I think it is only ever set to 'table' and we can more accurately check if it is a table by checking for columns/rows

Reply:
:exploding_head: 

Is there a way to make this less nested?